crates/egui/src/response.rs

```diff
@@ -224,7 +225,10 @@ impl Response {
             let pointer = &i.pointer;
 
             if pointer.any_click() {
-                if self.contains_pointer || self.hovered {
+                if self
+                    .flags
+                    .intersects(Flags::CONTAINS_POINTER | Flags::HOVERED)
+                {
```

Comment:
let's try to keep readability high and not make the code overly long

```suggestion
                if self.contains_pointer() || self.hovered() {
```